src/test/java/org/jitsi/testutils/ConfigPropertyTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.testutils;
+
+import com.typesafe.config.*;
+import org.jitsi.utils.config.*;
+
+import java.util.function.*;
+
+import static org.junit.Assert.assertEquals;
+
+public class ConfigPropertyTest<T extends ConfigProperty, U>
+{
+    public void runBasicTests(
+        String legacyPropName,
+        ParamResult<U> legacyParamResult,
+        String newPropName,
+        ParamResult<U> newParamResult,
+        Supplier<T> propInstanceSupplier
+    )
+    {
+        // With the provided arguments, we test each of the following scenarios:
+        // Old config is present, but doesn't provide a value
+        // Old config isn't present
+        // Old config is present and provides a value
+        // NOTE: it's assumed that a new config file (in the form of the defaults)
+        // will always be present, as it is bundled with the JVB itself
```

Comment:
It seems to me like it would be a good thing to test that the new config provides a value, to make sure that we don't forget to include it in the defaults file.

Reply:
well we're not using the actual config file here anyway, so those tests to me felt unnecessary.

Reply:
Hm, maybe not as part of this, but a separate test that the param has a default value would probably be a good thing to add to the standard set of tests.

Reply:
we'd have to load `reference.conf` manually, i think (as it wouldn't be on the classpath for the test by default, i think) and then we could probably do something like the [colibri resource test](https://github.com/jitsi/jitsi-videobridge/blob/master/src/test/java/org/jitsi/videobridge/rest/root/colibri/ColibriResourceTest.java) and find all the properties via reflection and then call `get` on them and make sure nothing throws.  i dunno if we'd get any false positives there, but could definitely try it out and see how it goes

Reply:
i'll take a shot at this in another PR